plugin/plugins/filesystem/nodes/pinfs.go

```diff
@@ -0,0 +1,93 @@
+package fsnodes
+
+import (
+	"context"
+	gopath "path"
+
+	"github.com/djdv/p9/p9"
+	logging "github.com/ipfs/go-log"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	coreoptions "github.com/ipfs/interface-go-ipfs-core/options"
+)
+
+type PinFS struct {
+	IPFSBase
+}
+
+//TODO: [review] check fields
+func PinFSAttacher(ctx context.Context, core coreiface.CoreAPI) *PinFS {
+	pd := &PinFS{IPFSBase: newIPFSBase(ctx, newRootPath("/ipfs"), p9.TypeDir,
+		core, logging.Logger("PinFS"))}
+	pd.meta, pd.metaMask = defaultRootAttr()
+	return pd
+}
+
+func (pd *PinFS) Attach() (p9.File, error) {
+	pd.Logger.Debugf("PD Attach")
+	//TODO: check core connection here
+	return pd, nil
+}
+
+func (pd *PinFS) GetAttr(req p9.AttrMask) (p9.QID, p9.AttrMask, p9.Attr, error) {
+	pd.Logger.Debugf("PD GetAttr")
+
+	return pd.Qid, pd.metaMask, pd.meta, nil
+}
+
+func (pd *PinFS) Walk(names []string) ([]p9.QID, p9.File, error) {
+	pd.Logger.Debugf("PD Walk names %v", names)
+	pd.Logger.Debugf("PD Walk myself: %v", pd.Qid)
+
+	if doClone(names) {
+		pd.Logger.Debugf("PD Walk cloned")
+		return []p9.QID{pd.Qid}, pd, nil
+	}
+
+	ipfsDir, err := IPFSAttacher(pd.Ctx, pd.core).Attach()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return ipfsDir.Walk(names)
+}
+
+func (pd *PinFS) Open(mode p9.OpenFlags) (p9.QID, uint32, error) {
+	pd.Logger.Debugf("PD Open")
+	return pd.Qid, 0, nil
+}
+
+func (pd *PinFS) Readdir(offset uint64, count uint32) ([]p9.Dirent, error) {
+	pd.Logger.Debugf("PD Readdir")
+
+	pins, err := pd.core.Pin().Ls(pd.Ctx, coreoptions.Pin.Type.Recursive())
+	if err != nil {
+		return nil, err
+	}
+
+	pLen := uint64(len(pins))
+	if offset >= pLen {
+		return nil, nil
+	}
+
+	//TODO: we should initialize and store entries during Open() to assure order is maintained through read calls
```

Comment:
:+1:.

Reply:
Resolved in https://github.com/ipfs/go-ipfs/pull/6612/commits/a2a9e8b6c57dcb783be95ef84329471ab913dc1a

Although I'm wondering if we ~micro-optimize~ eliminate the need for a subSlice with some index calculations.
`return directory.ents[offset:((len(ents)-offset) ... something with count)], nil`